content/lua/concepts/strings/terms/find/find.md

```diff
@@ -0,0 +1,176 @@
+---
+Title: '.find()'
+Description: 'The **`.find()`** function is used to search for a substring within a string and returns the start and end indices of the first occurrence of the substring.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+  - 'Game Development'
+Tags:
+  - 'Functions'
+  - 'Index'
+  - 'Methods'
+  - 'Search'
+  - 'Strings'
+CatalogContent:
+  - 'learn-lua'
+  - 'paths/computer-science'
+---
+
+The **`.find()`** function is used to search for a substring within a string and returns the start and end indices of the first occurrence of the substring.
+
+If the substring is not found, the function returns `nil`.
+
+## Syntax
+
+These are 2 different syntaxes for the .find() function:
+
+### 1
+
+```pseudo
+string.find(fullString, searchString, init, pattern)
+```
+
+This requires assigning the function to a variable, to return the function.
+
+or
+
+### 2
+
+```pseudo
+startindex, endindex = string.find(fullstring, searchstring, init, pattern)
+```
+
+This uses multiple assignment to save the results to an individual variables.
+
+- `fullString`: Is the string we are searching through.
+- `searchString`: Is the sub-string to search for.
+- `init` (optional): Is the initial index where the search starts. *Default index is 1*.
+- `pattern` (optional): Is a Boolean flag indicating whether to search using **pattern matching**[^1]. When set to false, pattern matching can be used, where as when set to true, pattern matching is ignored, and exact matches are searched for. *Default is false*.
+
+## Example 1
+
+Use `.find()` without multiple assignments:
```

Comment:
```suggestion
### Using `.find()` without multiple assignments
```
Let's change this to sub heading

Reply:
I can see why this is a change, I have basically already given all of the examples a sub-heading, so we might as well use them.